ElixrTalentConnect/Login/LoginViewController.swift

```diff
@@ -0,0 +1,162 @@
+//
+// LoginViewController.swift
+//  ElixrTalentConnect
+//
+//  Created by Devasurya on 10/01/24.
+//
+
+import UIKit
+import LocalAuthentication
+/// LoginViewcontroller contains  user authentication,sign-in ,sign-up.
+/// View in MVVM.
+class LoginViewController: UIViewController {
+    
+    /// Variables and constants  declaration
+    public  let userInfoPrompt = "Don't have an account? Sign up"
+    public  let textToChangeColor = " Sign up"
+    private var viewModel : LoginViewModel!
+    
+    /// Referencing  Outlets.
+    @IBOutlet weak var outerView: UIView!
+    @IBOutlet weak var signInButton: UIButton!
+    @IBOutlet weak var signInPromptLabel: UILabel!
+    @IBOutlet weak var emailField: UITextField!
+    @IBOutlet weak var passwordField: UITextField!
+    @IBOutlet weak var backGroundImage: UIImageView!
+    
+    /// View Life cycle.- Contain method calls to setup basic UI.
+    /// Setupcode to perform same action as that of IQkeyboard manager.
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        UISetup()
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+        viewModel = LoginViewModel()
+        self.navigationItem.setHidesBackButton(true, animated: true)
+        
+    }
+    
+    /// SignIntapped is the IBaction of the button "signInButton", which trigers alert actions,validation
+    /// - Parameter sender: UIbutton named "signInButton".
+    @IBAction func SignInTapped(_ sender: UIButton){
+        let loginModel = UserModel(userName: emailField.text , passwordValue:passwordField.text)
+        let validationResult = viewModel.validateCredentials(model: loginModel)
+        
+        if validationResult.isValid {
+            viewModel.authenticateWithBiometrics { [weak self] (success, error) in
+                if success {
+                    self!.navigateToHome()
+                    
+                } else {
+                    if let error = error {
+                        self!.showAlert(message: "Biometric authentication failed: \(error.localizedDescription)")
+                    }
+                }
+            }
+        } else {
+            showAlert(message: validationResult.message ?? "Invalid credentials.")
+        }
+    }
+        
+        /// Function to get the basic UI layout.
+        func UISetup() {
+            backGroundImage.layer.cornerRadius = 20
+            backGroundImage.clipsToBounds = true
+            signInPromptLabel.text = userInfoPrompt
+            let range = (userInfoPrompt as NSString).range(of: textToChangeColor)
+            let attributedText = NSMutableAttributedString.init(string: userInfoPrompt)
```

Comment:
why cant we use a button on top of label to get the tap effect. Do we really need to do this much?